### ISSUE TITLE
FunctionTransformer gets as input a numpy array rather than a series

### ISSUE THREAD

**Issue by hshteingart** [pandas-dev/sklearn-pandas#60]
If one want to use Series function on the input, one needs to transform the input back to pandas Series rather than just receiving it as pandas Series.
For example, say you would like a OneHotEncoding on the hour of a timestamp column. One would now need to do the following:

```
('timestamp', [
            FunctionTransformer(lambda x: pd.to_datetime(pd.Series(x)).apply(lambda t: t.hour).reshape(-1, 1),
                                validate=False), OneHotEncoder()])
```

Which is cumbursome.
If the dtype and Series object would have been kept it would be much shorter.

**Comment by dukebody** [comment]
sklearn-pandas passes a numpy array to sklearn transformers because this is what most transformers (if not all) expect. The fact that some of them can work with pandas dataframes or series is just because the pandas interface mimics the numpy array one.

Therefore I believe it's better if listed transformers always receive a numpy array. What do you think?

In the example you provide, I believe a cleaner (and possibly faster) solution would be the following:

```
df['hour'] = df['timestamp']
mapper = DataFrameMapper([
    (['hour'], OneHotEncoder())
])
```

Note that the use of brackets around the `hour` parameter selector allows passing a 2D numpy array to `OneHotEncoder`, which is what that transformer expects.

**Comment by jph00** [comment]
Do any transformers definitely not work with series? Because I came here to request the same thing - I actually created a little custom transformer for dates:

```
class DateEncoder():
    def fit(self, dat, y=None):
        return self
    
    def transform(self, dat):
        dt = dat.dt
        return [list(o) for o in zip(dt.year, dt.month, dt.day)]
```

...but it doesn't work for this reason. Creating extra columns seems unnecessarily cumbersome and memory intensive. Not a big deal I know, but just wanted to check this really isn't possible.

**Comment by dukebody** [comment]
@jph00 current version of sklearn-pandas converts selected columns to a numpy array before passing them to the transformers, so any transformers that need the input to be a Series or DataFrame won't work.

However I believe you can extract year, month and day from the numpy array using some of the solutions described at http://stackoverflow.com/q/13648774/356729

Perhaps we can add some attribute to the dataframe mapper definition to optionally not transform the columns to a numpy array for the first transformer, but keep them as a DataFrame or Series. What do you think?

**Comment by jph00** [comment]
That sounds like a good idea.

**Comment by dukebody** [comment]
Fixed in #85. Please reopen if it doesn't fit your use-case or see any bugs.
